<!-- Dataset HK_1 from UrbanNav project. -->
<launch>
    <!-- load dataset-specific params -->
    <rosparam command="load" file="$(find gnss_preprocessor)/config/hongkong.yaml" />

    <!-- path of dataset -->
    <param name="roverMeasureFile" type="string" value="$(find gnss_preprocessor)/dataset/UrbanNav/HK_1/COM3_190428_124409.obs" />
    <param name="baseMeasureFile" type="string" value="$(find gnss_preprocessor)/dataset/UrbanNav/HK_1/hksc1180.19o" />
    <param name="BeiDouEmpFile" type="string" value="$(find gnss_preprocessor)/dataset/UrbanNav/HK_1/hksc1180.19b" />
    <param name="GPSEmpFile" type="string" value="$(find gnss_preprocessor)/dataset/UrbanNav/HK_1/hksc1180.19n" />
    <param name="SP3file" type="string" value="$(find gnss_preprocessor)/dataset/UrbanNav/HK_1/ESA0MGNFIN_20191180000_01D_05M_ORB.SP3" />
    <param name="ionexFile" type="string" value="$(find gnss_preprocessor)/dataset/UrbanNav/HK_1/igsg1180.19i" />

    <param name="out_folder"      type="string" value="$(find gnss_preprocessor)/dataset/UrbanNav/HK_1/rtklibResult.pos" />

    <node name="gnss_preprocessor_node" pkg="gnss_preprocessor" type="gnss_preprocessor_node" output="screen" />
    

    <!-- Static fake TF transform -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="static_tf_fake" args="0 0 0 0 0 0 map base_link" />
    <node pkg="tf2_ros" type="static_transform_publisher" name="static_tf_fake_m2w" args="0 0 0 0 0 0 map world" />

    <!-- Rosbag recording -->
    <node pkg="rosbag" type="record" name="rosbag_HK_1" args="record -O $(find gnss_preprocessor)/dataset/Processed/UrbanNav_HK_1_preprocessed.bag -a"/> 

</launch>
